### ISSUE TITLE
Change to "Evil Visitor" behavior

### ISSUE THREAD

**Issue by BlackjackMKV** [ldtteam/minecolonies-features#498]
### Prerequisites 

- [x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x] I made sure that **this change request is not a duplicate of any existing change request.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.14.292-Alpha
- Related Mods and their Versions:

## Is your feature request related to a problem?

It's super annoying that some of the visitors just take your stuff and despawn when trying to recruit them. After all, if I wanted items randomly deleted, I could probably script it to do it straight from my inventory and save myself the hassle. On the other hand, while there is a config setting to change the likelihood or disable it entirely, I feel like it could be improved on instead.

## Enhancement or Change description

Maybe we could make it so that instead of simply despawning, they actually have to escape with the goods? Raids can tell where the approximate edges of the village are, so could it be that they have to actually run there before despawning? It would be super cool if you could actually try to kill them to get your stuff back if you're fast enough. Even better would be if guards could actually react to them trying to escape and either try to attack them like a hostile mob or, if you wanted to sink a LOT of work into it, maybe lock them in a new prison hut where you can keep them for some time. Maybe have them eventually convert to a citizen or just have the option to make them work in specific areas doing something?

I'd just be happy with being able to shoot them to get my stuff back though. They're not ninjas, they're just random visitors, so I kinda doubt they could realistically escape me when I have full Netherite gear, a speed potion, and elytra. :P

### Notes or related things

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by ravenbuilder934** [comment]
> Raids can tell where the approximate edges of the village are

as a note they actually don't. the raiders just spawn at least 20 blocks from a hut block

**Comment by BlackjackMKV** [comment]
> 
> 
> > Raids can tell where the approximate edges of the village are
> 
> as a note they actually don't. the raiders just spawn at least 20 blocks from a hut block

Yup, that's what I was referring to. Though, I suppose calling it understanding where the village ends was a bit ambiguous. The idea is basically that the visitor stealing your stuff has to make it at least that distance away before despawning so you have a chance to recoup your losses rather than just take the hit.
